src/OrchardCore/OrchardCore.Recipes.Abstractions/Services/IRecipeReader.cs

```diff
@@ -0,0 +1,13 @@
+using System.Threading.Tasks;
+using Microsoft.Extensions.FileProviders;
+using OrchardCore.Recipes.Models;
+
+namespace OrchardCore.Recipes.Services
+{
+    public interface IRecipeReader
+    {
+        Task<RecipeDescriptor> GetRecipeDescriptor(string recipeFilePath, IFileProvider fileProvider);
+
+        Task<RecipeDescriptor> GetRecipeDescriptor(string recipeFilePath, IFileInfo recipeFileInfo, IFileProvider fileProvider);
```

Comment:
Why this one?

Reply:
No wait, second one was not about optimization ... it was to support the harvester. I still need two methods ... one with recipeFilePath, and another with recipeFileInfo as first param. I'll make the change and you can review.

Reply:
Never mind all that ... I am keeping the second one and getting rid of the first one.